<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Sudarshan A. Shetty" />

<meta name="date" content="2017-12-30" />

<title>Microbiota anlaysis tutorial example</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.6/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.6/js/bootstrap.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
<script src="site_libs/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
<link href="site_libs/readthedown-0.1/readthedown.css" rel="stylesheet" />
<script src="site_libs/readthedown-0.1/readthedown.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="readthedown.css" type="text/css" />

</head>

<body>


<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Microbiome Tutorials</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Example_report.html">Microbiome Analysis (microbiomeutilities)</a>
</li>
<li>
  <a href="microbiome_tutorial.html">Microbiome Anlaysis Tutorial</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="content" data-toggle="wy-nav-shift">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->

<nav id="nav-top" role="navigation" aria-label="top navigation">
    <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
</nav>


<div id="header">
<h1 class="title">Microbiota anlaysis tutorial example</h1>
</div>


<div id="table-of-contents">
    <h2><a href="#content">Microbiota anlaysis tutorial example</a></h2>
    <div id="text-table-of-contents">
      <ul>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#example-data">Example data</a></li>
      <li><a href="#load-the-libraries">Load the libraries</a></li>
      <li><a href="#pre-processing-and-qc">Pre-processing and QC</a></li>
      <li><a href="#diversity">Diversity</a></li>
      <li><a href="#beta-diversity">Beta diversity</a></li>
      <li><a href="#quantifying-group-divergence-spread">Quantifying group divergence / spread</a></li>
      <li><a href="#composition">Composition</a></li>
      <li><a href="#boxplot">Boxplot</a></li>
      <li><a href="#core-microbiota">Core microbiota</a></li>
      <li><a href="#heatmap">Heatmap</a></li>
      <li><a href="#references">References</a></li>
      </ul>
    </div>
</div>

<div id="main">
<p>This tutorial gets you started with basic R tools for microbial ecology using an example data. In particular the aim is to provide an introduction to basic data handling, analysis and visulisation of marker gene amplicon sequencing data.</p>
<p>This is a supporting website for the main project:<br />
Tools for microbiome analysis in R. <strong>Microbiome package version 1.1.2.</strong> URL: <a href="http://microbiome.github.com/microbiome" class="uri">http://microbiome.github.com/microbiome</a>.</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>Launch R/RStudio and install the microbiome R package see (<a href="http://microbiome.github.io/microbiome/Installation.html">installation instructions</a>).</p>
<p>For instructions on how to load different file formats into R see <a href="http://microbiome.github.io/microbiome/Data.html">import data</a>.</p>
</div>
<div id="example-data" class="section level2">
<h2>Example data</h2>
<p>A subset of human intestinal biopsy and faecal data from pre-print <a href="https://www.biorxiv.org/content/early/2017/12/06/227272">Shetty SA et al. 2017</a>.</p>
<p>This is a simplified version of various methods available these days to microbial ecologists. The ideology of putting all of this together is to share the information and also clarify the ‘ease’(you see we didn’t say ‘simple’) of using R-software and related packages. The analyses shown here are basic and aimed mostly at introducing the reader to commonly used packages, scripts and data analysis methods. Descision making related to different parameters will still be soley upon the user.<br />
For more information you can have a look at <a href="http://joey711.github.io/phyloseq/">Phyloseq</a> and <a href="http://joey711.github.io/phyloseq-demo/phyloseq-demo.html">here</a>.</p>
<p>Important Note 1:<br />
Wisdom1 - There is no substitute for careful reading, so read the tutorial first and then start playing with it.<br />
Wisdom2 - Never skip a step or piece of text, you might need a file that was generated previously. Take your time and don’t rush.</p>
<p>Kindly cite all the packages/tools that you have used in your analysis.</p>
</div>
<div id="load-the-libraries" class="section level2">
<h2>Load the libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome)
<span class="kw">library</span>(knitr)
<span class="kw">library</span>(ggpubr)
<span class="kw">library</span>(reshape2)
<span class="kw">library</span>(RColorBrewer)
<span class="kw">library</span>(microbiomeutilities)
<span class="kw">library</span>(viridis)
<span class="kw">library</span>(tibble)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;biogeogut&quot;</span>)
<span class="co">#save(biogeogut, file = &quot;./inputdata/biogeogut.rda&quot;)</span>
<span class="co">#load(&quot;./inputdata/biogeogut.rda&quot;)</span>
ps1 &lt;-<span class="st"> </span>biogeogut</code></pre></div>
</div>
<div id="pre-processing-and-qc" class="section level2">
<h2>Pre-processing and QC</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># check the data</span>

<span class="kw">print</span>(ps1)</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 19063 taxa and 14 samples ]
## sample_data() Sample Data:       [ 14 samples by 6 sample variables ]
## tax_table()   Taxonomy Table:    [ 19063 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># check if any OTUs are not present in any samples</span>
<span class="kw">any</span>(<span class="kw">taxa_sums</span>(ps1) ==<span class="st"> </span><span class="dv">0</span>)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>The answer is TRUE. Yes, there are OTUs not found in any samples. This is usually the case when data is subset to remove some samples. OTUs unique to those sample are not removed along with the samples. Therefore, it is important to check this everytime the phyloseq object is filtered for samples using <code>subset_samples</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1a &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(ps1) &gt;<span class="st"> </span><span class="dv">0</span>, ps1)

<span class="co"># check again if any OTUs are not present in any samples</span>
<span class="kw">any</span>(<span class="kw">taxa_sums</span>(ps1a) ==<span class="st"> </span><span class="dv">0</span>)</code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>After the <code>prune_taxa</code> function is run to remove OTus with zero occurances, the answer is “FALSE”.</p>
<p>Check how many OTUs are kept.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># subtract the number of OTUs in original (ps1) with number of OTUs in new phyloseq (ps1a) object.</span>
<span class="co"># no. of OTUs in original </span>
<span class="kw">ntaxa</span>(ps1)</code></pre></div>
<pre><code>## [1] 19063</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># no. of OTUs in new </span>
<span class="kw">ntaxa</span>(ps1a)</code></pre></div>
<pre><code>## [1] 10947</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ntaxa</span>(ps1) -<span class="st"> </span><span class="kw">ntaxa</span>(ps1a)</code></pre></div>
<pre><code>## [1] 8116</code></pre>
<p>8116 OTUs have been removed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Check the taxonomy levels</span>

<span class="kw">rank_names</span>(ps1a)</code></pre></div>
<pre><code>## [1] &quot;Domain&quot;  &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;   &quot;Species&quot;</code></pre>
<p>Check distribution of how many reads/samples?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">SeqDepth =<span class="st"> </span><span class="kw">colSums</span>(<span class="kw">otu_table</span>(ps1a))
<span class="kw">sample_data</span>(ps1a)$SeqDepth =<span class="st"> </span>SeqDepth
<span class="kw">qplot</span>(<span class="kw">log10</span>(SeqDepth), <span class="dt">geom =</span> <span class="st">&quot;histogram&quot;</span>) +<span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-6-1.png" width="768" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We use the main variable here of interest SampleType</span>

<span class="kw">ggplot</span>(<span class="kw">meta</span>(ps1a)) +
<span class="st">    </span><span class="kw">geom_histogram</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">log10</span>(SeqDepth)), <span class="dt">alpha=</span> <span class="fl">0.6</span>) +<span class="st"> </span><span class="kw">facet_wrap</span>(~SampleType) +<span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-6-2.png" width="768" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sort</span>(SeqDepth)</code></pre></div>
<pre><code>##  cb126  cb169  cb110  cs164  cs165  cs114  cs110  cs169  cs116  cb114 
## 175365 185945 194901 197472 206850 211300 217896 227476 253549 265298 
##  cb116  cs126  cb165  cb164 
## 281864 298923 335089 452182</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># min number of reads</span>
<span class="kw">min</span>(SeqDepth)</code></pre></div>
<pre><code>## [1] 175365</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># max number of reads</span>
<span class="kw">max</span>(SeqDepth)</code></pre></div>
<pre><code>## [1] 452182</code></pre>
<p>Also just plot number of reads per sample.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">meta</span>(ps1a))</code></pre></div>
<pre><code>##       SampleID     Subject Age InputFileName Sex SampleType SeqDepth
## cb110    cb110 Subject_110   C   cb110.fasta   1     Biopsy   194901
## cb126    cb126 Subject_126   C   cb126.fasta   2     Biopsy   175365
## cb114    cb114 Subject_114   A   cb114.fasta   1     Biopsy   265298
## cb116    cb116 Subject_116   D   cb116.fasta   1     Biopsy   281864
## cb165    cb165 Subject_165   A   cb165.fasta   1     Biopsy   335089
## cb169    cb169 Subject_169   C   cb169.fasta   2     Biopsy   185945</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggbarplot</span>(<span class="kw">meta</span>(ps1a), <span class="st">&quot;SampleID&quot;</span>, <span class="st">&quot;SeqDepth&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;SampleType&quot;</span>) +<span class="st"> </span><span class="kw">rotate_x_text</span>()</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-7-1.png" width="768" /></p>
<p>It is clear that there is a difference in the sequencing depth. Keep this in mind for downstream analysis.<br />
If you have installed microbiome package from github then you can simple run the following command.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summarize_phyloseq</span>(ps1a)</code></pre></div>
<pre><code>## Compositional = NO
## 1] Min. number of reads = 175365 
## 2] Max. number of reads = 452182 
## 3] Total number of reads = 3504110 
## 4] Average number of reads = 250293.571428571 
## 5] Median number of reads = 222686 
## 7] Sparsity = 0.746257944120372 
## 6] Any OTU sum to 1 or less? YES 
## 8] Number of singletons = 2643 
## 9] Percent of OTUs that are singletons 24.1436009865717 
## 10] Number of sample variables are: 7 
## SampleID 
## Subject 
## Age 
## InputFileName 
## Sex 
## SampleType 
## SeqDepth</code></pre>
<p>We have data which is not relative abundance, only counts in our case. sparsity is how populated is the data with zeros. There are also singletons, 24% of the total OTUs are observed only once. Depending on your research question, you can choose to remove them form further analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># remove singletons</span>

ps1b &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(ps1a) &gt;<span class="st"> </span><span class="dv">1</span>, ps1a)

ps1b</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 8304 taxa and 14 samples ]
## sample_data() Sample Data:       [ 14 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 8304 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summarize_phyloseq</span>(ps1b)</code></pre></div>
<pre><code>## Compositional = NO
## 1] Min. number of reads = 175066 
## 2] Max. number of reads = 451671 
## 3] Total number of reads = 3501467 
## 4] Average number of reads = 250104.785714286 
## 5] Median number of reads = 222595.5 
## 7] Sparsity = 0.68823114505918 
## 6] Any OTU sum to 1 or less? NO 
## 8] Number of singletons = 0 
## 9] Percent of OTUs that are singletons 0 
## 10] Number of sample variables are: 7 
## SampleID 
## Subject 
## Age 
## InputFileName 
## Sex 
## SampleType 
## SeqDepth</code></pre>
<p>After removing singletons, there are 8304 OTUs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># check for distribution of OTUs </span>

<span class="kw">hist</span>(<span class="kw">log10</span>(<span class="kw">taxa_sums</span>(ps1b)))</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-10-1.png" width="768" /></p>
<p>The data is left tailed. Common for microbiome count data. We can also look at how prevalent are OTUs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prev.otu &lt;-<span class="st"> </span><span class="kw">plot_taxa_prevalence</span>(ps1b, <span class="st">&quot;Phylum&quot;</span>)

<span class="kw">print</span>(prev.otu)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-11-1.png" width="1152" /></p>
<p>We can see that there are OTUs that are very low abundance and present in only few samples.<br />
We will remove OTUs unclassified at phylum level</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps2 &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(ps1b, Phylum !=<span class="st"> &quot;p__&quot;</span>)

<span class="co"># unique(tax_table(ps2)[,&quot;Phylum&quot;])</span>

ps2a &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(ps2,Class!=<span class="st">&quot;Chloroplast&quot;</span>)

ps3 &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(ps2a,Order!=<span class="st">&quot;Mitochondria&quot;</span>)

ps3</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 8298 taxa and 14 samples ]
## sample_data() Sample Data:       [ 14 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 8298 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="diversity" class="section level2">
<h2>Diversity</h2>
<p>Of specific interest in microbial ecology is the diversity of microbial communities.</p>
<p>First we will calculate eveness. We will use unfiltered data with singletons.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#check which indices are available by simply typying the following and enter</span>

<span class="co"># ?evenness</span>

<span class="co"># there are several options</span>

ps.even &lt;-<span class="st"> </span><span class="kw">evenness</span>(ps1a, <span class="dt">index =</span> <span class="st">&quot;all&quot;</span>) 

<span class="kw">kable</span>(<span class="kw">head</span>(ps.even))</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">camargo</th>
<th align="right">pielou</th>
<th align="right">simpson</th>
<th align="right">evar</th>
<th align="right">bulla</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cb110</td>
<td align="right">0.0156946</td>
<td align="right">0.5034986</td>
<td align="right">0.0009180</td>
<td align="right">0.2389022</td>
<td align="right">0.0890463</td>
</tr>
<tr class="even">
<td>cb126</td>
<td align="right">0.0332422</td>
<td align="right">0.6348806</td>
<td align="right">0.0028337</td>
<td align="right">0.2404437</td>
<td align="right">0.1329307</td>
</tr>
<tr class="odd">
<td>cb114</td>
<td align="right">0.0262603</td>
<td align="right">0.6275053</td>
<td align="right">0.0032954</td>
<td align="right">0.2075605</td>
<td align="right">0.1156905</td>
</tr>
<tr class="even">
<td>cb116</td>
<td align="right">0.0184672</td>
<td align="right">0.5791351</td>
<td align="right">0.0027806</td>
<td align="right">0.2110912</td>
<td align="right">0.0951103</td>
</tr>
<tr class="odd">
<td>cb165</td>
<td align="right">0.0326793</td>
<td align="right">0.6405342</td>
<td align="right">0.0044317</td>
<td align="right">0.2065294</td>
<td align="right">0.1288252</td>
</tr>
<tr class="even">
<td>cb169</td>
<td align="right">0.0132030</td>
<td align="right">0.5744764</td>
<td align="right">0.0022607</td>
<td align="right">0.2082446</td>
<td align="right">0.0767695</td>
</tr>
</tbody>
</table>
<p><strong>Plot</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1a.meta &lt;-<span class="st"> </span><span class="kw">meta</span>(ps1a)

ps1a.meta$simpson &lt;-<span class="st"> </span>ps.even$simpson </code></pre></div>
<p>Before doing any statistical test check for the distribution of the diversity.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(ps1a.meta$simpson)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-15-1.png" width="768" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># This data look ok</span>

<span class="co"># test is it is</span>
<span class="kw">shapiro.test</span>(ps1a.meta$simpson)</code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  ps1a.meta$simpson
## W = 0.92008, p-value = 0.2205</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">qqnorm</span>(ps1a.meta$simpson)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-15-2.png" width="768" /></p>
<p>If you have more than two variable with a category then do the following. If only two variable like in this case Biopsy and faeces skip this chunk.<br />
See also <a href="http://microbiome.github.io/microbiome/PlotDiversity.html">main tutorial</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># create a list of pairwise comaprisons</span>
smtype &lt;-<span class="st"> </span><span class="kw">levels</span>(ps1a.meta$SampleType) <span class="co"># get the variables</span>

<span class="co"># make a pairwise list that we want to compare.</span>
smtype.pairs &lt;-<span class="st"> </span><span class="kw">combn</span>(<span class="kw">seq_along</span>(smtype), <span class="dv">2</span>, <span class="dt">simplify =</span> <span class="ot">FALSE</span>, <span class="dt">FUN =</span> function(i)smtype[i])

<span class="kw">print</span>(smtype.pairs)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span><span class="kw">ggviolin</span>(ps1a.meta, <span class="dt">x =</span> <span class="st">&quot;SampleType&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;simpson&quot;</span>,
 <span class="dt">add =</span> <span class="st">&quot;boxplot&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;SampleType&quot;</span>, 
 <span class="dt">palette =</span> <span class="kw">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>),
 <span class="dt">legend =</span> <span class="st">&quot;right&quot;</span>) 

<span class="kw">print</span>(p1)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-17-1.png" width="768" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span>p1 +<span class="st"> </span><span class="kw">stat_compare_means</span>(<span class="dt">method =</span> <span class="st">&quot;t.test&quot;</span>)

<span class="kw">print</span>(p1)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-17-2.png" width="768" /></p>
<p>For more information and useful tips and suggestions check the <a href="http://www.sthda.com/english/rpkgs/ggpubr/">Statistical tools for high-throughput data analysis</a>.</p>
<p>For more option on diversity indices <a href="http://microbiome.github.io/microbiome/Diversity.html">Diversity</a></p>
</div>
<div id="beta-diversity" class="section level2">
<h2>Beta diversity</h2>
<p>Now we will check similarities in microbial community. Distance based matrices such as <a href="https://www.sciencedirect.com/science/article/pii/S0065250408601683">Bray-Curtis</a>, <a href="http://aem.asm.org/content/71/12/8228.full">Unifrac</a> are commonly used and visualized using <a href="http://ordination.okstate.edu/overview.htm">ordination methods</a>.</p>
<p>We will remove some potential suporious OTUs from data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nsamples</span>(ps3)</code></pre></div>
<pre><code>## [1] 14</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># we will keep only those OTUs that are detected alteast 5 times in 5 out of total 14 samples</span>
ps4 &lt;-<span class="st"> </span><span class="kw">core</span>(ps3, <span class="dt">detection =</span> <span class="dv">5</span>, <span class="dt">prevalence =</span> <span class="dv">5</span>/<span class="kw">nsamples</span>(ps3))
<span class="kw">hist</span>(<span class="kw">log10</span>(<span class="kw">taxa_sums</span>(ps4)))</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-18-1.png" width="768" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps4</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 897 taxa and 14 samples ]
## sample_data() Sample Data:       [ 14 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 897 taxa by 7 taxonomic ranks ]</code></pre>
<p>Since, there is no phylogenetic tree for this data set, Bray-Curtis distance will be calculated.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps4.rel &lt;-<span class="st"> </span>microbiome::<span class="kw">transform</span>(ps4, <span class="st">&quot;compositional&quot;</span>)

bx.ord_pcoa_bray &lt;-<span class="st"> </span><span class="kw">ordinate</span>(ps4.rel, <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;bray&quot;</span>)

<span class="co">#Scree plot</span>
<span class="kw">plot_scree</span>(bx.ord_pcoa_bray) +<span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-19-1.png" width="768" /></p>
<p>Axis 1 and 2 are of interest.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">beta.ps1 &lt;-<span class="st"> </span><span class="kw">plot_ordination</span>(ps4.rel, 
                            bx.ord_pcoa_bray, 
                            <span class="dt">color=</span><span class="st">&quot;Subject&quot;</span>, 
                            <span class="dt">label =</span> <span class="st">&quot;Subject&quot;</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">shape =</span> SampleType), <span class="dt">size=</span> <span class="dv">4</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="dv">0</span>, <span class="dt">size =</span> <span class="dv">12</span>))</code></pre></div>
<pre><code>## Warning: Ignoring unknown aesthetics: na.rm</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">beta.ps1 &lt;-<span class="st"> </span>beta.ps1 +<span class="st"> </span><span class="kw">theme_bw</span>(<span class="dt">base_size =</span> <span class="dv">14</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())

<span class="co"># Now we can join the biopsy and stool from same subject</span>
beta.ps2 &lt;-<span class="st"> </span>beta.ps1 +<span class="st"> </span><span class="kw">geom_line</span>() +<span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Dark2&quot;</span>)
beta.ps2</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-20-1.png" width="768" /></p>
<p>This ordination plot shows that axis 1 explains 36% of the variation between biopsy and stool.</p>
<p>For asthetics, an ellipse can be added.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">beta.ps3 &lt;-<span class="st"> </span><span class="kw">plot_ordination</span>(ps4.rel, 
                            bx.ord_pcoa_bray, 
                            <span class="dt">color=</span><span class="st">&quot;SampleType&quot;</span>, 
                            <span class="dt">label =</span> <span class="st">&quot;Subject&quot;</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span> <span class="dv">4</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="dv">0</span>, <span class="dt">size =</span> <span class="dv">12</span>))</code></pre></div>
<pre><code>## Warning: Ignoring unknown aesthetics: na.rm</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">beta.ps3 &lt;-<span class="st"> </span>beta.ps3 +<span class="st"> </span><span class="kw">theme_bw</span>(<span class="dt">base_size =</span> <span class="dv">14</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())

beta.ps3 +<span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Dark2&quot;</span>) +<span class="st"> </span><span class="kw">stat_ellipse</span>()</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-21-1.png" width="768" /></p>
<p><strong>PERMANOVA</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">metadf.bx &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(ps4.rel))
bray_ps.bxn &lt;-<span class="st"> </span>phyloseq::<span class="kw">distance</span>(<span class="dt">physeq =</span> ps4.rel, <span class="dt">method =</span> <span class="st">&quot;bray&quot;</span>)

<span class="kw">set.seed</span>(<span class="dv">995</span>)
<span class="co"># Adonis test</span>
<span class="kw">library</span>(vegan)</code></pre></div>
<pre><code>## Warning: package &#39;vegan&#39; was built under R version 3.4.3</code></pre>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.4-5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">adonis.test &lt;-<span class="st"> </span><span class="kw">adonis</span>(bray_ps.bxn ~<span class="st"> </span>SampleType, <span class="dt">data =</span> metadf.bx)
adonis.test</code></pre></div>
<pre><code>## 
## Call:
## adonis(formula = bray_ps.bxn ~ SampleType, data = metadf.bx) 
## 
## Permutation: free
## Number of permutations: 999
## 
## Terms added sequentially (first to last)
## 
##            Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)   
## SampleType  1    1.4675 1.46753  6.5019 0.35142  0.003 **
## Residuals  12    2.7085 0.22571         0.64858          
## Total      13    4.1760                 1.00000          
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Adonis test shows that the difference is significant and 35% variation is explained by SampleType.</p>
<p><strong>Checking the homogeneity condition</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Note the assumption of similar multivariate spread among the groups</span>
<span class="co"># ie. analogous to variance homogeneity</span>
<span class="co"># Here the groups have signif. different spreads and</span>
<span class="co"># permanova result may be potentially explained by that.</span>
dist &lt;-<span class="st"> </span><span class="kw">vegdist</span>(<span class="kw">t</span>(<span class="kw">abundances</span>(ps4.rel)))
<span class="kw">anova</span>(<span class="kw">betadisper</span>(dist, metadf.bx$SampleType))</code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: Distances
##           Df  Sum Sq  Mean Sq F value  Pr(&gt;F)  
## Groups     1 0.11638 0.116379  3.5611 0.08357 .
## Residuals 12 0.39217 0.032681                  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>betadisper test supports that the sampletype explains major differfences in overall microbiota composition.</p>
</div>
<div id="quantifying-group-divergence-spread" class="section level2">
<h2>Quantifying group divergence / spread</h2>
<p>Divergence of a given sample set can be quantified as the average dissimilarity of each sample from the group mean; the dissimilarity can be quantified by beta diversity, for instance. This was applied in group-level comparisons for instance in <a href="https://www.nature.com/articles/ismej201463">Salonen et al. ISME J 2014</a>. They focused on homogeneity using inverse correlation, whereas here we focus on divergence using correlation but the measure is essentially the same. For more information, check <a href="http://microbiome.github.io/microbiome/Betadiversity.html">Beta diversity and microbiome divergence</a></p>
<p>Calculate group divergences within the Biopsy and Stool samples</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">b.st &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">divergence</span>(<span class="kw">subset_samples</span>(ps4, SampleType ==<span class="st"> &quot;Stool&quot;</span>)))
b.bx &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">divergence</span>(<span class="kw">subset_samples</span>(ps4, SampleType ==<span class="st"> &quot;Biopsy&quot;</span>)))</code></pre></div>
<p>Plot the divergence</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">div_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(b.st, b.bx)
<span class="kw">colnames</span>(div_df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Biopsy&quot;</span>)

dif.g &lt;-<span class="st"> </span>reshape2::<span class="kw">melt</span>(div_df)</code></pre></div>
<pre><code>## No id variables; using all as measure variables</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ggpubr::<span class="kw">ggboxplot</span>(dif.g, <span class="st">&quot;variable&quot;</span>, <span class="st">&quot;value&quot;</span>, 
                  <span class="dt">ylab =</span> <span class="st">&quot;Divergence&quot;</span>, 
                  <span class="dt">xlab =</span> <span class="st">&quot;Sample Type&quot;</span>, 
                  <span class="dt">add =</span> <span class="st">&quot;jitter&quot;</span>,
                  <span class="dt">fill =</span> <span class="st">&quot;variable&quot;</span>,
                  <span class="dt">palette =</span> <span class="kw">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>))</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-25-1.png" width="768" /></p>
<p>The biopsies tend to have smaller values, indicating that the samples are more similar to the group mean, and the biopsy is less heterogeneous (has smaller spread / is more homogeneous).</p>
</div>
<div id="composition" class="section level2">
<h2>Composition</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1a.com &lt;-<span class="st"> </span>ps1a

taxic &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(ps1a.com@tax_table) 

<span class="co"># Add the OTU ids from OTU table into the taxa table at the end.</span>
taxic$OTU &lt;-<span class="st"> </span><span class="kw">rownames</span>(taxic) 

<span class="co"># You can see that we now have extra taxonomy levels.</span>
<span class="kw">colnames</span>(taxic)</code></pre></div>
<pre><code>## [1] &quot;Domain&quot;  &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;   &quot;Species&quot;
## [8] &quot;OTU&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># convert it into a matrix.</span>
taxmat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(taxic)

<span class="co"># convert into phyloseq compaitble file.</span>
new.tax &lt;-<span class="st"> </span><span class="kw">tax_table</span>(taxmat)  

<span class="co"># incroporate into phyloseq Object</span>
<span class="kw">tax_table</span>(ps1a.com) &lt;-<span class="st"> </span>new.tax </code></pre></div>
<p><em>Phylum level</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pseq.ph &lt;-<span class="st"> </span><span class="kw">aggregate_taxa</span>(ps1a.com, <span class="st">&quot;Phylum&quot;</span>, <span class="dt">top =</span> <span class="dv">11</span>)

p.phy &lt;-<span class="st"> </span><span class="kw">plot_composition</span>(pseq.ph, <span class="dt">sample.sort =</span> <span class="ot">NULL</span>, <span class="dt">otu.sort =</span> <span class="ot">NULL</span>,
  <span class="dt">x.label =</span> <span class="st">&quot;SampleType&quot;</span>, <span class="dt">plot.type =</span> <span class="st">&quot;barplot&quot;</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)

<span class="kw">print</span>(p.phy +<span class="st"> </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>) +<span class="st"> </span><span class="kw">theme_bw</span>())</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-27-1.png" width="768" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># it would be nice to have the Taxonomic names in italics.</span>
<span class="co"># for that we set this</span>
guide_italics &lt;-<span class="st"> </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">label.theme =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">15</span>, 
    <span class="dt">face =</span> <span class="st">&quot;italic&quot;</span>, <span class="dt">colour =</span> <span class="st">&quot;Black&quot;</span>, <span class="dt">angle =</span> <span class="dv">0</span>)))

pseq.ph.rel &lt;-<span class="st"> </span>microbiome::<span class="kw">transform</span>(pseq.ph, <span class="st">&quot;compositional&quot;</span>)


plot.comp.rel &lt;-<span class="st"> </span><span class="kw">plot_composition</span>(pseq.ph.rel, <span class="dt">x.label =</span> <span class="st">&quot;SampleType&quot;</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) +<span class="st"> </span><span class="kw">theme_bw</span>() +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Relative abundance&quot;</span>) +<span class="st"> </span>guide_italics +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">18</span>))

plot.comp.rel +<span class="st"> </span><span class="kw">scale_fill_brewer</span>( <span class="st">&quot;Phylum&quot;</span>,<span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-28-1.png" width="768" /></p>
</div>
<div id="boxplot" class="section level2">
<h2>Boxplot</h2>
<p>Check which are the top ten genera and how they differ between sample types.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pn &lt;-<span class="st"> </span><span class="kw">plot_taxa_boxplot</span>(ps4.rel, <span class="st">&quot;Genus&quot;</span>, <span class="dv">10</span>, <span class="st">&quot;SampleType&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;Set2&quot;</span>, <span class="st">&quot;Relative abundance of top 10 genera&quot;</span>)</code></pre></div>
<pre><code>## The phy_tree slot is empty, easy to make the plot</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(pn)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-29-1.png" width="768" /></p>
</div>
<div id="core-microbiota" class="section level2">
<h2>Core microbiota</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.bx &lt;-<span class="st"> </span><span class="kw">subset_samples</span>(ps1a, SampleType ==<span class="st"> &quot;Biopsy&quot;</span>)

ps1.bx.rel &lt;-<span class="st"> </span>microbiome::<span class="kw">transform</span>(ps1.bx, <span class="st">&quot;compositional&quot;</span>)
<span class="co">#your original pseq/relative abundance file</span>
<span class="co">#if</span>
<span class="kw">colnames</span>(<span class="kw">tax_table</span>(ps1.bx.rel))</code></pre></div>
<pre><code>## [1] &quot;Domain&quot;  &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;   &quot;Species&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#last column has SVs/OTU ids then you can skip the following five steps and go to aggregate_taxa step.</span>

taxic &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(ps1.bx.rel@tax_table)
taxic$OTU &lt;-<span class="st"> </span><span class="kw">row.names</span>(taxic)
<span class="co">#convert it into a matrix.</span>
taxmat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(taxic)

<span class="co">#convert into phyloseq compaitble file.</span>
new.tax &lt;-<span class="st"> </span><span class="kw">tax_table</span>(taxmat)

<span class="co">#incroporate into phyloseq Object</span>
<span class="kw">tax_table</span>(ps1.bx.rel) &lt;-<span class="st"> </span>new.tax

<span class="co">#the presence of NA is an issue.</span>

<span class="kw">tax_table</span>(ps1.bx.rel)[,<span class="st">&quot;Genus&quot;</span>][<span class="kw">is.na</span>(<span class="kw">tax_table</span>(ps1.bx.rel)[,<span class="st">&quot;Genus&quot;</span>])] &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">tolower</span>(<span class="kw">substring</span>(<span class="st">&quot;Genus&quot;</span>, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="st">&quot;__&quot;</span>)

<span class="co">#at family level</span>
ps1.bx.gen &lt;-<span class="st"> </span><span class="kw">aggregate_taxa</span>(ps1.bx.rel, <span class="st">&quot;Genus&quot;</span>)

<span class="co">#Set different detection levels and prevalence</span>
prevalences &lt;-<span class="st"> </span><span class="kw">seq</span>(.<span class="dv">5</span>, <span class="dv">1</span>, .<span class="dv">5</span>) <span class="co">#0.5 = 95% prevalence</span>
detections &lt;-<span class="st"> </span><span class="dv">10</span>^<span class="kw">seq</span>(<span class="kw">log10</span>(<span class="fl">1e-3</span>), <span class="kw">log10</span>(.<span class="dv">2</span>), <span class="dt">length =</span> <span class="dv">10</span>)
<span class="co">#(1e-3) = 0.001% abundance; change &quot;-3&quot; to -2 to increase to 0.01%</span>

p &lt;-<span class="st"> </span><span class="kw">plot_core</span>(ps1.bx.gen, <span class="dt">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, 
               <span class="dt">colours =</span> <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">10</span>, <span class="st">&quot;Spectral&quot;</span>)),
               <span class="dt">min.prevalence =</span> <span class="fl">0.9</span>, 
               <span class="dt">prevalences =</span> prevalences, 
               <span class="dt">detections =</span> detections) +
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)
<span class="kw">print</span>(p)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-30-1.png" width="576" /></p>
</div>
<div id="heatmap" class="section level2">
<h2>Heatmap</h2>
<p>We have data in counts, hence transformation to log10(x+1) will be done for better plotting. Plot top 20 OTUs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.c &lt;-<span class="st"> </span><span class="kw">format_to_besthit</span>(ps1a)

heat.sample &lt;-<span class="st"> </span><span class="kw">plot_taxa_heatmap</span>(ps1a, <span class="dt">subset.top =</span> <span class="dv">20</span>,
    <span class="dt">VariableA =</span> <span class="st">&quot;SampleType&quot;</span>,
    <span class="dt">heatcolors =</span> <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">100</span>, <span class="st">&quot;Blues&quot;</span>)),
    <span class="dt">transformation =</span> <span class="st">&quot;log10&quot;</span>)</code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-31-1.png" width="1152" /></p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<ol style="list-style-type: decimal">
<li><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2674678/">Phylogenetic Diversity</a></p></li>
<li><p><a href="http://aem.asm.org/content/71/12/8228.full">UniFrac</a></p></li>
<li><p><a href="http://www2.ib.unicamp.br/profs/thomas/NE002_2011/maio10/Magurran%202004%20c2-4.pdf">General Diversity</a></p></li>
<li><p><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003531">To rarify or not to rarefy</a></p></li>
<li><p><a href="hhttp://msystems.asm.org/content/2/1/e00127-16">Microbiome Helper</a></p></li>
<li><p><a href="http://parfreylab.botany.ubc.ca/pca-pcoa-and-nmds/">Ordinations</a></p></li>
<li><p><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">Phyloseq paper</a></p></li>
<li><p><a href="https://f1000research.com/articles/5-1492/v1">Phyloseq workflow</a></p></li>
<li><p><a href="https://rdrr.io/cran/RAM/">R for Amplicon-Sequencing-Based Microbial-Ecology</a></p></li>
<li><p><a href="https://github.com/Lagkouvardos/Rhea">Rhea</a></p></li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.1 (2017-06-30)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 16299)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] vegan_2.4-5               lattice_0.20-35          
##  [3] permute_0.9-4             bindrcpp_0.2             
##  [5] tibble_1.3.4              viridis_0.4.0            
##  [7] viridisLite_0.2.0         microbiomeutilities_0.1.1
##  [9] RColorBrewer_1.1-2        reshape2_1.4.3           
## [11] ggpubr_0.1.6              magrittr_1.5             
## [13] knitr_1.17                microbiome_1.1.2         
## [15] ggplot2_2.2.1             phyloseq_1.20.0          
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.36.2      tidyr_0.7.2         jsonlite_1.5       
##  [4] splines_3.4.1       foreach_1.4.3       shiny_1.0.5        
##  [7] assertthat_0.2.0    highr_0.6           stats4_3.4.1       
## [10] yaml_2.1.14         ggrepel_0.7.0       backports_1.1.1    
## [13] glue_1.1.1          digest_0.6.12       XVector_0.16.0     
## [16] colorspace_1.3-2    htmltools_0.3.6     httpuv_1.3.5       
## [19] Matrix_1.2-11       plyr_1.8.4          pkgconfig_2.0.1    
## [22] pheatmap_1.0.8      questionr_0.6.1     bookdown_0.5       
## [25] zlibbioc_1.22.0     purrr_0.2.3         xtable_1.8-2       
## [28] scales_0.5.0        mgcv_1.8-22         IRanges_2.10.4     
## [31] BiocGenerics_0.22.0 lazyeval_0.2.0      survival_2.41-3    
## [34] mime_0.5            evaluate_0.10.1     nlme_3.1-131       
## [37] MASS_7.3-47         tools_3.4.1         data.table_1.10.4-3
## [40] stringr_1.2.0       S4Vectors_0.14.6    munsell_0.4.3      
## [43] cluster_2.0.6       Biostrings_2.44.2   ade4_1.7-8         
## [46] compiler_3.4.1      rlang_0.1.2.9000    rhdf5_2.20.0       
## [49] grid_3.4.1          iterators_1.0.8     biomformat_1.4.0   
## [52] rstudioapi_0.7      igraph_1.1.2        miniUI_0.1.1       
## [55] labeling_0.3        rmarkdown_1.6       gtable_0.2.0       
## [58] codetools_0.2-15    multtest_2.32.0     R6_2.2.2           
## [61] gridExtra_2.3       dplyr_0.7.4         bindr_0.1          
## [64] rprojroot_1.2       ape_5.0             stringi_1.1.5      
## [67] parallel_3.4.1      rmdformats_0.3.3    Rcpp_0.12.13       
## [70] tidyselect_0.2.0</code></pre>
<p>Kindly cite this work as follows: “Leo Lahti, Sudarshan Shetty et al. (2017). Tools for microbiome analysis in R. Version 1.1.2. URL: <a href="http://microbiome.github.com/microbiome" class="uri">http://microbiome.github.com/microbiome</a>. Check also the relevant references listed in the manual page of each function.</p>
<p>The package utilizes tools from a number of other R extensions, including dplyr (Wickham, Francois, Henry, et al., 2017), ggplot2 (Wickham, 2009), phyloseq (McMurdie and Holmes, 2013), tidyr (Wickham, 2017), vegan (Oksanen, Blanchet, Friendly, et al., 2017).</p>
<p>This website theme was created by modifiying the rmdformats readthedown format.</p>
</div>
</div>


</div>

<div id="postamble" data-toggle="wy-nav-shift" class="status">
<p class="author"><span class="glyphicon glyphicon-user"></span> Sudarshan A. Shetty</p>
<p class="date"><span class="glyphicon glyphicon-calendar"></span> 2017-12-30</p>
</div>


<script>
$(document).ready(function () {
 	 	$('#content img')
 	  .addClass("image-thumb");
      $('#content img')
 	  .addClass("image-lb");
  $('#content').magnificPopup({
	      type:'image',
	      closeOnContentClick: false,
	      delegate: 'img',
	      gallery: {enabled: true },
	      image: {
	        verticalFit: true,
          titleSrc: 'alt'
	      }
 	    });
 	});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
